netlogo-gui/src/main/lab/gui/LabManager.scala

```diff
@@ -36,7 +39,20 @@ class LabManager(val workspace:        GUIWorkspace,
     protocols.clear()
   }
   private lazy val dialog = new ManagerDialog(this, dialogFactory, menuFactory)
-  def show() { dialog.update(); dialog.setVisible(true) }
+  def updateDialogCoordinates(): Unit = {
+    //If parent window lies in the same place, do not update coordinates
+    val parentLocation = workspace.getFrame.getLocationOnScreen
+    if(currentParentLocation!=null && currentParentLocation.x==parentLocation.x && parentLocation.y==parentLocation.y)
+      return
+    //Set new coordinates
+    currentParentLocation = parentLocation
```

Comment:
If I comment out L43-L48, the dialog appears in the right place without "flickering." Can you test whether commenting them out removes the flicker for you? If it does, let's consider them unnecessary and remove them.

Reply:
Ohkay, right! L49 was sufficient. I've removed the function, and added the needed code to show() function